### ISSUE TITLE
How to pretrain "Predictor"

### ISSUE THREAD

**Issue by Dahee96** [Crystalsound/FRN#1]
Hi,
I want to use your code for PLC train challenge data.
However, since that data consists of 16kHz, the pretrained "Predictor" cannot be used.
I need to pretrain the "Predictor" but don't know how to train it.

Could you explain in detail?
Input: the magnitude of the previous frame
Model: only Predictor ("Joiner" is omitted)
Output: the magnitude of the current frame

Is it right? 
Then, what did you use as a loss function to train the "Predictor" only?

Thanks.

**Comment by vietanh125** [comment]
Hi,

We didn't include predictor training scripts for a clean repo (they are almost like 2 independent implementation). However, if you are interested of pretraining the predictor, here is some details you can consider:

- The predictor takes magnitude of the current frame and outputs magnitude of the next frame. To create the input, you can right-shift the magnitude sequence and left-pad with a zero vector; and the target is just the magnitude of the stft.

> STFT: [frame 0, frame 1, frame 2,...,frame n]
> Input: [0, mag 0, mag 1,..., mag n-1]
> Output: [mag 0, mag 1, mag 2,..., mag n]

- We used MSE loss for the predictor. Learning rate is 1e-4, max batch size is 150 but it should converge after about 20 epochs.
- You should create another project for pretraining utilizing the FRN template so that the main project won't be messy.

Finally, you can skip the pretraining step, just initialize the predictor without loading the weight and train with the entire architecture simultaneously. The result would only be slightly different.
